tests/syntax/test_modular.py

```diff
@@ -809,6 +811,91 @@ def test_override_behavior():
     assert tuple(actions) == (1, -1, -2, 2)
 
 
+def test_override_none_behavior():
+    scenario = compileScenic(
+        """
+        scenario Main():
+            setup:
+                ego = new Object
+            compose:
+                wait
+                do Sub() for 2 steps
+                wait
+        scenario Sub():
+            setup:
+                override ego with behavior Bar
+        behavior Bar():
+            x = -1
+            while True:
+                take x
+                x -= 1
+        """,
+        scenario="Main",
+    )
+    actions = sampleEgoActions(scenario, maxSteps=4)
+    assert tuple(actions) == (None, -1, -2, None)
+
+
+def test_override_leakage():
+    scenario = compileScenic(
+        """
+        scenario Main():
+            setup:
+                ego = new Object with prop 1
+            compose:
+                do Sub1()
+        scenario Sub1():
+            setup:
+                override ego with prop 2
+                override ego with behavior Bar
+        behavior Bar():
+            terminate
+        """,
+        scenario="Main",
+    )
+    scene, _ = generateChecked(scenario, 1)
+    assert scene.objects[0].prop == 1
+    sampleActionsFromScene(
+        scene,
+        maxIterations=1,
+        maxSteps=1,
+        singleAction=True,
+        asMapping=False,
+        timestep=1,
+    )
```

Comment:
```suggestion
    sampleActionsFromScene(scene)
```
(Writing out all the parameters is confusing since you don't actually change any of them from their default values. The defaults are set to be what we want for most tests, and only the ones which need to be changed are written out in other tests.)